src/main/java/Hamyo.java

```diff
@@ -0,0 +1,103 @@
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.Scanner;
+
+import hamyo.misc.HamyoException;
+import hamyo.misc.Parser;
+import hamyo.misc.Storage;
+import hamyo.misc.Ui;
+import hamyo.tasks.TaskList;
+
+/**
+ * Main class of the Hamyo Chatbot.
+ *
+ * @author Han Yu.
+ */
+public class Hamyo {
+
+    private final Ui ui;
+    private final Storage storage;
+    private final TaskList tasks;
+    private boolean isActive;
+
+    /**
+     * Constructor of Hamyo instance. Consists of Ui for user interactions,
+     * TaskList to store users' tasks, Storage to load and save users' list of
+     * tasks.
+     */
+    public Hamyo() {
+        this.ui = new Ui();
+        this.tasks = new TaskList();
+        this.isActive = true;
+        this.storage = new Storage("./savedTasks.txt");
+
+        this.ui.greet();
+    }
+
+    /**
+     * Continuously scans for user input and execute commands while it is
+     * active, and terminates once this.active is set to false using the bye
+     * command.
+     */
+    public void run() {
```

Comment:
Maybe refactor the nested try-catch blocks by separating to a method for error handling, ensuring that each block of functionality is more isolated and manageable.